node/full.go

```diff
@@ -341,10 +341,12 @@ func (n *FullNode) OnStart() error {
 		go n.blockManager.BlockSubmissionLoop(n.ctx)
 		go n.headerPublishLoop(n.ctx)
 		go n.blockPublishLoop(n.ctx)
+	} else {
+		go n.blockManager.RetrieveLoop(n.ctx)
+		go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
+		go n.blockManager.SyncLoop(n.ctx, n.cancel)
 	}
-	go n.blockManager.RetrieveLoop(n.ctx)
-	go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
-	go n.blockManager.SyncLoop(n.ctx, n.cancel)
+
```

Comment:
Returning from the if condition reads slightly better:

```go
	if n.nodeConfig.Aggregator {
		n.Logger.Info("working in aggregator mode", "block time", n.nodeConfig.BlockTime)
		go n.blockManager.AggregationLoop(n.ctx, n.nodeConfig.LazyAggregator)
		go n.blockManager.BlockSubmissionLoop(n.ctx)
		go n.headerPublishLoop(n.ctx)
		go n.blockPublishLoop(n.ctx)
		return nil
	}

	go n.blockManager.RetrieveLoop(n.ctx)
	go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
	go n.blockManager.SyncLoop(n.ctx, n.cancel)

	return nil
```

Reply:
true